### PR TITLE
Changes for the Network_compliance_workflow_manager

### PR DESCRIPTION
**Pull request by sonalideepthi777**
## Description
Update the Network complaince workflow module to support for replacement of membership APIs, and  improved functionality with version-based routing.

## Type of Change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [x] Tasks are idempotent (can be run multiple times without changing state)
- [ ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [x] Playbooks are modular and reusable
- [x] Handlers are used for actions that need to run on change

## Documentation
- [x] All options and parameters are documented clearly.
- [x] Examples are provided and tested.
- [x] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/network_compliance_workflow_manager.py [hunk 1]
```diff
@@ -613,7 +626,7 @@ def get_device_ids_from_site(self, site_name, site_id):
         Retrieves the management IP addresses and their corresponding instance UUIDs of devices associated with a specific site in Cisco Catalyst Center.
 
         Parameters:
-            site_name (str): The name of the site whose devices" information is to be retrieved.
+            site_name (str): The name of the site whose devices' information is to be retrieved.
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:629]: Many modules using get_device_ids_from_site.. Can we put into dnac.py under sites section?

./swim_workflow_manager.py
./site_workflow_manager.py
./site_intent.py
./network_compliance_workflow_manager.py
./accesspoint_workflow_manager.py
./device_configs_backup_workflow_manager.py
./swim_intent.py
./provision_workflow_manager.py

File: plugins/modules/network_compliance_workflow_manager.py [hunk 2]
```diff
@@ -1098,27 +1133,50 @@ def get_task_status(self, task_id, task_name):
             If an error occurs during the API call, it will be caught and logged.
         """
         # Make an API call to retrieve the task tree
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:1135]: This get_task_status also used in many modules.. Can we write in dnac.py and make use of it?

**Review by madhansansel — APPROVED** [review]

File: plugins/modules/network_compliance_workflow_manager.py (hunk 2 above)
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Modified

File: plugins/modules/network_compliance_workflow_manager.py (hunk 1 above)
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: This function added inside the dnac.py

File: plugins/modules/network_compliance_workflow_manager.py [hunk 3]
```diff
@@ -505,48 +527,39 @@ def validate_run_compliance_paramters(self, mgmt_ip_to_instance_id_map, run_comp
     def site_exists(self, site_name):
```
[the remark this answers is not in the document]
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Removed this API

File: plugins/modules/network_compliance_workflow_manager.py [hunk 4]
```diff
@@ -354,6 +371,11 @@ def __init__(self, module):
           The method does not return a value.
         """
         super().__init__(module)
+        self.payload = module.params
+        self.keymap = {}
+        self.dnac_version = int(self.payload.get(
```
[the remark this answers is not in the document]
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Removed this part

**Review by madhansansel — APPROVED** [review]

File: plugins/modules/network_compliance_workflow_manager.py [hunk 5]
```diff
@@ -1261,7 +1030,7 @@ def get_task_result(self, task_id, device_ids):
         start_time = time.time()
 
         while True:
-            response = self.get_task_status(task_id, task_name)
+            response = self.get_task_status_from_tasks_by_id(task_id, task_name)
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:1033]: In the latest dnac.py, it is defined as:
 def get_task_status_from_tasks_by_id(self, task_id, task_name, success_msg):

Can you check?

Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Based on the def get_task_status_from_tasks_by_id(self, task_id, task_name, success_msg) function, the get_task_result function in the network_compliance_module has been modified.

File: plugins/modules/network_compliance_workflow_manager.py [hunk 6]
```diff
@@ -1259,37 +1028,43 @@ def get_task_result(self, task_id, device_ids):
...
-            # Check if task failed
-            if "failed" in response.get("progress").lower():
+            # Task failed
+            if "failed" in response.get("progress", "").lower():
                 device_ids_str = ", ".join(device_ids)
                 msg = "Failed to {0} on the following device(s): {1}".format(task_name, device_ids_str)
                 return {task_id: {"msg": msg, "status": "Failed"}}
 
+            # Optional: sleep to avoid hammering the API
+            time.sleep(5)
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:1066]: We should not use any hard-coded value..

File: plugins/modules/network_compliance_workflow_manager.py [hunk 7]
```diff
@@ -1259,37 +1028,43 @@ def get_task_result(self, task_id, device_ids):
         """
```
Comment (madhansansel) [inline plugins/modules/network_compliance_workflow_manager.py:1028]: This API also should be in dnac.py.. Can you please check? We already have APIs in dnac.py. If not, we have to move this API in dnac.py.

File: plugins/modules/network_compliance_workflow_manager.py (hunk 6 above)
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Removed

File: plugins/modules/network_compliance_workflow_manager.py (hunk 7 above)
Comment (sonalideepthi777) [reply plugins/modules/network_compliance_workflow_manager.py]: Moved to dnac.py

**Pull request merged**